src/main/java/api/StudentApi.java

```diff
@@ -5,38 +5,71 @@
 
 import com.google.gson.Gson;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import model.Lesson;
 import model.Student;
+import model.Subject;
+import model.Teacher;
 import persistence.StudentPersistence;
+import persistence.SubjectPersistence;
+import persistence.TeacherPersistence;
 import spark.Request;
 import spark.Response;
 
 public class StudentApi {
   private static final Gson gson = new Gson();
-  private static final StudentPersistence persistence = new StudentPersistence();
+  private final StudentPersistence studentPersistence;
+  private final TeacherPersistence teacherPersistence;
+  private final SubjectPersistence subjectPersistence;
+
+  public StudentApi(
+      StudentPersistence studentPersistence,
+      TeacherPersistence teacherPersistence,
+      SubjectPersistence subjectPersistence) {
+    this.studentPersistence = studentPersistence;
+    this.teacherPersistence = teacherPersistence;
+    this.subjectPersistence = subjectPersistence;
+  }
+
+  public List<api.dto.Student> getAll(Request request, Response response) {
+    return convertToDto(studentPersistence.getAll());
+  }
 
-  public List<Student> getAll(Request request, Response response) {
-    return persistence.getAll();
+  private List<api.dto.Student> convertToDto(List<Student> students) {
+    return students.stream().map((Student student) -> api.dto.Student.fromStudent(student, subjectPersistence)).collect(Collectors.toList());
   }
 
-  public List<Student> getAllFilteredByTeacher(Request request, Response response) {
-    String teacherId = request.queryParams("teacherId");
-    return persistence.getAll();
+  public List<api.dto.Student> getAllByTeacherAndSubject(Request request, Response response) {
```

Comment:
Importy got bzz

Reply:
??????????????????????????????????????????????

Reply:
i mean api.dto.Student xd

Reply:
zmienił bym nazwę na StudentDto czy coś

Reply:
żeby rozróżnienie było

Reply:
pełna nazwa pakietowa jest wystarczająca, wskazuje ona na to "rozróżnienie"